client/lib/signup/private-by-default.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:signup:private-by-default' );
+
+interface PrivateByDefaultSiteSettings {
+	flowName?: string;
+}
+
+/**
+ * Should the site be private by default
+ * @param siteSettings PrivateByDefaultSiteSettings
+ * @returns `true` for private by default & `false` for not
+ */
+export function shouldBePrivateByDefault( {
+	flowName = '',
```

Comment:
_Minor_

Rather than replacing `undefined` with `''` (empty string), what do you think of letting the `undefined` case be handled with the match?

```suggestion
	flowName,
```

and below

```ts
	if ( flowName && flowName.match( /^ecommerce/ ) ) { /* … */ }
```